packages/notifications/push/amplify_push_notifications_pinpoint/test/pinpoint_provider_test.dart

```diff
@@ -70,6 +70,45 @@ void main() {
       );
     });
 
+    test('identifyUser should throw exception if the underneath call throws',
+        () async {
+      when(mockAmplifyAuthProviderRepository.getAuthProvider(any))
+          .thenReturn(awsIamAmplifyAuthProvider);
+      when(
+        mockAnalyticsClient.init(
+          pinpointAppId: anyNamed('pinpointAppId'),
+          region: anyNamed('region'),
+          authProvider: anyNamed('authProvider'),
+        ),
+      ).thenAnswer((realInvocation) async {});
+
+      final mockEndpointClient = MockEndpointClient();
+
+      when(
+        mockAnalyticsClient.endpointClient,
+      ).thenReturn(mockEndpointClient);
+
+      await pinpointProvider.init(
+        config: notificationsPinpointConfig,
+        authProviderRepo: mockAmplifyAuthProviderRepository,
+        analyticsClient: mockAnalyticsClient,
+      );
+      when(mockEndpointClient.setUser(any, any)).thenThrow(Exception());
+      expect(
+        pinpointProvider.identifyUser(
+          userId: 'userId',
+          userProfile: MockUserProfile(),
+        ),
+        throwsA(
+          isA<PushNotificationException>().having(
+            (e) => e.message,
+            'Unable to identify user',
+            contains('Unable to identify user.'),
+          ),
+        ),
+      );
+    });
+
     test(
```

Comment:
Can you add a test to check for the presence of the header? Here's an example: https://github.com/aws-amplify/amplify-flutter/blob/next/packages/authenticator/amplify_authenticator/example/integration_test/http_test.dart